ComponentKit/Components/CKTestTypedPropsComponent.mm

```diff
@@ -0,0 +1,42 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#import "CKTestTypedPropsComponent.h"
+
+#import "CKTypedPropsComponentSubclass.h"
+#import "CKLabelComponent.h"
+
+@implementation CKTestTypedPropsComponent
+
++ (instancetype)newWithProps:(const CKTestTypedPropsComponentProps &)props
+                        view:(const CKComponentViewConfiguration &)view
+                        size:(const CKComponentSize &)size
+{
+  return [self newWithPropsStruct:props
+                             view:view
+                             size:size];
+}
+
++ (CKComponent *)renderWithProps:(const CKTypedComponentStruct &)p
```

Comment:
The subclass should implement this method, but is forced to unwrap the props. I don't think that's *too* big of a deal since it's internal to the component, and we currently do this for id type to state already, so it's not a dealbreaker for me.